lib/ReactViews/Clipboard.tsx

```diff
@@ -24,57 +23,51 @@ interface ClipboardProps {
 }
 
 const Clipboard: React.FC<ClipboardProps> = (props) => {
-  const { id, source, theme, rounded } = props;
+  const { id, source, theme, rounded, text, onCopy } = props;
   const { t } = useTranslation();
   const [status, setStatus] = useState<CopyStatus>(
     CopyStatus.NotCopiedOrWaiting
   );
-  useEffect(() => {
-    // Setup clipboard.js and show a tooltip on copy success or error for 3s
-    const clipboardBtn = new clipboard(`.btn-copy-${id}`);
-    let timerId: ReturnType<typeof setTimeout> | null = null;
-    function removeTimeout() {
-      if (timerId !== null) {
-        clearTimeout(timerId);
-        timerId = null;
+
+  const handleCopy = async () => {
+    try {
+      if (text) {
+        await navigator.clipboard.writeText(text);
+        setStatus(CopyStatus.Success);
+        if (onCopy) onCopy(text);
+      } else {
+        setStatus(CopyStatus.Error);
       }
+    } catch {
+      setStatus(CopyStatus.Error);
     }
-    function resetTooltipLater() {
-      removeTimeout();
-      timerId = setTimeout(() => {
+  };
+
+  useEffect(() => {
+    // Setup clipboard.js and show a tooltip on copy success or error for 3s
```

Comment:
If you're removing clipboard.js, I think it makes sense to update the comments so clipboard.js isn't mentioned anywhere.

(There's another mention of clipboard.js on line 13 in this file.)